src/framework/scene-registry.js

```diff
@@ -347,6 +359,41 @@ class SceneRegistry {
         });
     }
 
+    /**
+     * Change to a new scene. Calling this function will load the scene data, delete all
+     * entities and graph nodes under `app.root` and load the scene settings and hierarchy.
+     *
+     * @param {SceneRegistryItem | string} sceneItem - The scene item (which can be found with
+     * {@link SceneRegistry#find}, URL of the scene file (e.g."scene_id.json") or name of the scene.
+     * @param {ChangeSceneCallback} [callback] - The function to call after loading,
+     * passed (err, entity) where err is null if no errors occurred.
+     * @example
+     * app.scenes.ChangeScene("Scene Name", function (err, entity) {
+     *     if (!err) {
+     *         // success
+     *     } else {
+     *         // error
+     *     }
+     * });
+     */
+    changeScene(sceneItem, callback) {
+        const app = this._app;
+
+        const onBeforeAddHierarchy = (sceneItem) => {
```

Comment:
This `app` closure variable also has too much context and maybe add JSDoc?

Reply:
Is JSDoc needed here for a private function that is only used in the scope of the current class?

Reply:
I guess its not needed, I just use them as a mind crutch and when the types are checked, it can offer some insight about what is going wrong in which position